e2e/tests/overview-page.spec.ts

```diff
@@ -33,15 +33,17 @@ test('matches the screenshot', async ({page}) => {
 test('shows an error that their query is invalid', async ({page}) => {
   await page.goto('http://localhost:5555/?q=available_on%3Achrom');
 
-  await page.locator('.message').waitFor({state: 'visible'});
-  // The following works in --ui mode, but fails in batch mode.
-  // expect(page.locator('.message')).toContainText('Invalid query...');
+  await page.waitForSelector('.message:has-text("Invalid query...")', {
+    state: 'visible',
+  });
+  const pageContainer = page.locator('.page-container');
+  await expect(pageContainer).toHaveScreenshot('invalid-query.png');
 });
 
 test('shows an unknown error when there is an internal error', async ({
   page,
 }) => {
-  await page.route('**/v1/features', route =>
+  await page.route('**/v1/features?page_size=25', route =>
```

Comment:
We needed to add the query parameter used. It was not matching.

Reply:
Ah, I'm glad you looked into this.